asf_core_data/pipeline/data_joining/merge_install_dates.py

```diff
@@ -23,118 +24,74 @@ def get_mcs_install_dates(epc_df, additional_features=False):
         pandas.DataFrame: EPC dataset with added MCS install dates.
     """
 
-    # Note
-    # TO DO:  This will be simplified shortly to work with UPRN instead of address
+    newest_joined_batch = data_batches.get_latest_mcs_epc_joined_batch()
 
-    # Get original address from EPC
-    epc_df["original_address"] = (
-        epc_df["ADDRESS1"] + epc_df["ADDRESS2"] + epc_df["POSTCODE"]
-    )
-    epc_df["original_address"] = (
-        epc_df["original_address"]
-        .str.strip()
-        .str.lower()
-        .replace(r"\s+", "", regex=True)
-    )
-
-    # Load MCS
+    # # Load MCS
     mcs_data = data_getters.load_s3_data(
         base_config.BUCKET_NAME,
-        base_config.MCS_EPC_MERGED_PATH,
+        newest_joined_batch,
         usecols=[
-            "date",
+            "commission_date",
             "tech_type",
-            "compressed_epc_address",
-            "address_1",
-            "address_2",
-            "address_3",
-            "postcode",
             "version",
-            "alt_type",
+            "tech_type",
             "installation_type",
-            "# records",
+            "UPRN",
+            "cluster",
+            "installer_name",
+            "postcode",
         ],
+        dtype={"UPRN": "str"},
     )
 
-    mcs_data.fillna({"address_1": "", "address_2": "", "address_3": ""}, inplace=True)
-
     # Rename columns
     mcs_data.rename(
-        columns={
-            "date": "HP_INSTALL_DATE",
-            "tech_type": "Type of HP",
-            "compressed_epc_address": "compressed_epc_address",
-            "address_1": "MCS address 1",
-            "address_2": "MCS address 2",
-            "address_3": "MCS address 3",
-            "postcode": "MCS postcode",
-        },
+        columns={"commission_date": "HP_INSTALL_DATE"},
         inplace=True,
     )
 
-    # Get original EPC address from MCS/EPC match
-    mcs_data = mcs_data.loc[~mcs_data["compressed_epc_address"].isna()]
-    mcs_data["MCS_ADDRESS"] = (
-        mcs_data["MCS address 1"]
-        + " "
-        + mcs_data["MCS address 2"]
-        + " "
-        + mcs_data["MCS address 3"]
-        + " "
-        + mcs_data["MCS postcode"]
-    )
-
-    mcs_data["compressed_epc_address"] = (
-        mcs_data["compressed_epc_address"]
-        .str.strip()
-        .str.lower()
-        .replace(r"\s+", "", regex=True)
-    )
-
     # Get the MCS install dates
     mcs_data["HP_INSTALL_DATE"] = pd.to_datetime(
         mcs_data["HP_INSTALL_DATE"]
         .str.strip()
         .str.lower()
+        .replace(r"\s.*", "", regex=True)
```

Comment:
can't this regex and the one in the line below be put together?

Reply:
Done.